src/annotator-content-analysis.js

```diff
@@ -1,54 +1,136 @@
+        this.setData(this.options.data);
+      } else {
+        let data = this.getUrl(this.options.dataUrl);
+        data.then((res) => {
+          this.setData(res)
+        });
+      }
+    },
+
+    // shortcut for AJAX get
+    getUrl: function(url) {
+      return $.get(url, (data) => {
+        return data;
+      });
     },
 
-    // method to model our raw data payload
+    // TODO: make this more universal
+    //
+    // method to cast our data payload into module's context
     // @param value [object] API payload returned via AJAX
     //
-    configureData: function(value){
-      // TODO: establish how we want to configure the data
+    setData: function(value){
       this.data = {
         next: value.next,
         previous: value.previous,
-        glossary: value.results[0].analysis_type.glossary
+        analysis_type: value.results[0].analysis_type
       }
     },
 
-    // hooks into Annotator lifecycle events
+    // returns a promise that isn't resolved until our survey is submitted
     //
-    // TODOS:
-    //  • how to create our own UI
-    //  • how best to store our answers to interact with other plugins
-    //  • how to pause the annotator lifecycle?
-    beforeAnnotationCreated: function(annotation){
-      return this.promiseHelper(annotation)
-        .then((value) => {
-          this.setUpCustomEditorPanel(value);
-        });
+    createSurvey: function(annotation) {
+      return new Promise((resolve, reject) => {
+        this.data.analysis_type.topics.forEach(this.getTopics.bind(this));
+        let data = {
+          thisTopics: this.survey.topics.join('')
+        }
+
+        let formTemplate = Handlebars.compile(this.templates.form)
+
+        $('body').append(formTemplate(data))
+
+        $('.submit').on('click', (e) => {
+          console.log($('form').serialize())
+          resolve(annotation)
+          return false
+        })
+
+      })
     },
 
-    setUpCustomEditorPanel: function(annotation) {
-      console.log(annotation);
+    getTopics: function(element, index, array) {
+      element.questions.forEach(this.getQuestions.bind(this));
+      let data = {
+        element: element,
+        thisQuestions: this.survey.questions.join('')
+      }
+      this.survey.questions = []
+      let topicTemplate = Handlebars.compile(this.templates.topic);
+
+      this.survey.topics.push(topicTemplate(data))
+    },
+
+    getQuestions: function(element, index, array) {
+      element.answers.forEach(this.getAnswers.bind(this, element.id));
+      let data = {
+        element: element,
+        thisAnswers: this.survey.answers.join('')
+      }
+      this.survey.answers = []
+      let questionTemplate = Handlebars.compile(this.templates.question);
+      this.survey.questions.push(questionTemplate(data))
     },
 
-    // helper promise function for intercepting Annotator UI
+    getAnswers: function(parentId, element, index, array) {
+      let data = {
+        element: element,
```

Comment:
you can use the ES6 whatever-its-called notation: 

```
let data = { element, parentId }
```

Also I think it's considered good practice to use `const` instead of `let` unless you intend to mutate the variable